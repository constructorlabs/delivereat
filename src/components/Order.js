import React from "react";
import OrderItem from './OrderItem'
import '../styles/Order.scss';


function Order({newOrder, menu, orderTotals, sendOrder, customer, handleClickLogout, handleClickLogin}){
      return (
        <div className="orders">
        <ul className="order__items">
            {Object.values(newOrder).map(orderItem => {
                return <OrderItem menuItem={menu[orderItem.id]} orderItem={orderItem} key={orderItem.id} />
            })}
        </ul>
        <p><strong>{orderTotals.deliveryMessage}</strong></p>
        <p>Discount :{!orderTotals.discount?"10% on order more than £50":`£${orderTotals.discount.toFixed(2)}`}</p>
        <p>Pizzas : £{orderTotals.itemsCost}</p>
        <p>Delivery : £{orderTotals.deliveryCost} </p>
        <p></p>
        <p><strong>Total :£{orderTotals.itemsCost?orderTotals.itemsCost + orderTotals.deliveryCost - orderTotals.discount:0} </strong></p>
        {!!customer.name === true && (
          <p>Logged in as {customer.name} <button onClick={handleClickLogout}>Logout</button></p>
        )}
        {!customer.name === true && (
          <p><button id="login" onClick={handleClickLogin}>Login / Register</button></p>
        )}
        


        {orderTotals.itemsCost > 1 && (      
          <button id="order" onClick={sendOrder}>ORDER ME PIZZA!!!</button>
        )}

        </div>
      )
    }  

export default Order;