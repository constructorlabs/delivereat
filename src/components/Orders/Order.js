import React from 'react';

class Order extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            quantity: 0
        }

        this.ordersHandler = this.ordersHandler.bind(this);
        this.increaseQuantityHandler = this.increaseQuantityHandler.bind(this);
        this.decreaseQuantityHandler = this.decreaseQuantityHandler.bind(this);
    }

    ordersHandler(dishId, quantity, price, action) {
        this.props.receiver(dishId, quantity, price, action);
    }

    increaseQuantityHandler(dishId, price, action) {
        const qty = this.state.quantity + 1;
        this.setState({
            quantity: qty
        })
        this.ordersHandler(dishId, qty, price, action)
    }

    decreaseQuantityHandler(dishId, price, action) {
        if (this.state.quantity > 0) {
            const qty = this.state.quantity - 1;
            this.setState({
                quantity: qty
            })
            this.ordersHandler(dishId, qty, price, action)
        }
    }

    componentWillReceiveProps(props) {
        this.setState({ quantity: props.qty });
    }

    render() {
        const { dishId, price, qty } = this.props;
        const order = {};
        Object.keys(this.props.menu).forEach(orderKey => {
            if (this.props.menu[orderKey].id === dishId) {
                Object.assign(order, this.props.menu[orderKey])
            }
        })
        return (
            <div className="dish">
                <div className="dish__image" style={{ backgroundImage: `url(${order.image})` }}></div>
                <div className="dish__details">
                    <h2 className="dish__heading">{order.name}</h2>
                    <p className="dish__ingredients">{order.ingredients}.</p>
                    <span className="dish__price"><strong>Price: </strong>&pound;{order.price}</span>
                    <button onClick={(e) => this.decreaseQuantityHandler(order.id, order.price, "decrease")}
                        id="dish__quantity-decrease"
                        className={"dish__quantity-decrease " + (qty > 0 ? '' : 'disabled')}>â€“</button>
                    <button onClick={(e) => this.increaseQuantityHandler(order.id, order.price, "increase")}
                        id="dish__quantity-increase"
                        className="dish__quantity-increase">+</button>
                    <div id="dish__quantity-details"
                        className={"dish__quantity-details " + (qty > 0 ? '' : 'disabled')}>
                        {qty} in the basket.
                        </div>
                </div>
            </div>
        )
    }
}

export default Order;
