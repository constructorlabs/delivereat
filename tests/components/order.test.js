import React from 'react';
import { shallow } from 'enzyme';

import Order from '../../src/components/Order.js';
import renderer from 'react-test-renderer';

const newOrder = {}
const menu = {}
const customer = {}

describe('order', ()=> {
    it('rendered output snapshot', ()=> {
        const orderTotals = {itemsCost: 0, deliveryCost: 5, discount: 0, deliveryMessage: "Free delivery on orders over £30"}
          const tree = renderer.create(<Order orderTotals={orderTotals} newOrder={newOrder} customer={customer} menu={menu} />).toJSON()
          expect(tree).toMatchSnapshot()
    })
    it('when login clicked it should call handleClickLogin', ()=> {
        const orderTotals = {itemsCost: 0, deliveryCost: 5, discount: 0, deliveryMessage: "Free delivery on orders over £30"}
        const handleClickLogin= jest.fn();
        const wrapper = shallow(<Order orderTotals={orderTotals} newOrder={newOrder} customer={customer} menu={menu} handleClickLogin={handleClickLogin}/>);
        wrapper.find('#login').simulate('click');
        expect(handleClickLogin).toHaveBeenCalled()
    })
    it('when order button clicked it should call sendOrder', ()=> {
        const orderTotals = {itemsCost: 5, deliveryCost: 5, discount: 0, deliveryMessage: "Free delivery on orders over £30"}
        const sendOrder = jest.fn();
        const wrapper = shallow(<Order orderTotals={orderTotals} newOrder={newOrder} customer={customer} menu={menu} sendOrder={sendOrder}/>);
        wrapper.find('#order').simulate('click');
        expect(sendOrder).toHaveBeenCalled()
    })
})