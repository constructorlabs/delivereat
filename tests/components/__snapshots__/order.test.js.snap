// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order rendered output snapshot 1`] = `
<div
  className="orders"
>
  <ul
    className="order__items"
  />
  <p>
    <strong>
      Free delivery on orders over £30
    </strong>
  </p>
  <p>
    Discount :
    10% on order more than £50
  </p>
  <p>
    Pizzas : £
    0
  </p>
  <p>
    Delivery : £
    5
     
  </p>
  <p />
  <p>
    <strong>
      Total :£
      0
       
    </strong>
  </p>
  <p>
    <button
      id="login"
      onClick={undefined}
    >
      Login / Register
    </button>
  </p>
</div>
`;
